### ISSUE TITLE
[ENH] XLSTM in sktime

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#6793]
Add xlstm in sktime: https://github.com/muslehal/xLSTMTime

**Comment by Sohaib-Ahmed21** [comment]
I have started research on this issue and will try to open draft PR soon.

**Comment by Sohaib-Ahmed21** [comment]
Due to some complexity of this issue, I am leaving this and this is open to work. I'm starting my contribution in sktime with https://github.com/sktime/sktime/issues/6831

**Comment by fkiraly** [comment]
can you comment on this, @phoeenniixx , so I can assign you?

**Comment by phoeenniixx** [comment]
yeah please assign this issue to me @fkiraly

**Comment by vedantag17** [comment]
Hi, is this issue still available? I’m interested in working on it.

**Comment by fkiraly** [comment]
Yes, it is available - I believe @phoeenniixx has a PR in `pytorch-forecasting` with xlstm, but that is parked due to interface compatibility issues with the rest of `pytorch-forecasting`, if I remember correctly.

@phoeenniixx, please correct me if I am wrong!

**Comment by phoeenniixx** [comment]
Yes, https://github.com/sktime/pytorch-forecasting/pull/1709 is currently on hold as we're focusing on the v2.0. There were also some issues with the implementation, largely due to my limited understanding of `ptf` at the time.
And I think once we have a complete working v2.0, that would make the API more general and consistent, and we can directly move `xlstm` to `ptf` then. Adding this model to v1 could make migration to v2 hectic as we already have a lot of models that we have to migrate, we can atleast leave this one and just implement directly in a more stable v2 :)

**Comment by phoeenniixx** [comment]
Although for `sktime`,  I am not quite sure if a standalone implementation here would be helpful or a wrapper around the `ptf` implementation... as `ptf`'s implementation (if we wait till v2.0 and then I complete my PR) would take some time.

**Comment by vedantag17** [comment]
Thanks for the clarification, @phoeenniixx and @fkiraly!

In that case, would it make sense for me to wait until the ptf v2.0 stabilizes and the xlstm PR is finalized, so I can contribute a wrapper for sktime around the updated ptf implementation?
Or, alternatively, if there’s any scope for a temporary standalone implementation in sktime for experimentation or benchmarking purposes, I’d be happy to explore that route too.
Let me know what would be more useful to the project right now!

**Comment by fkiraly** [comment]
> if there’s any scope for a temporary standalone implementation in sktime for experimentation or benchmarking purposes, I’d be happy to explore that route too.

I think there is scope for that - I do not think it hurts, it might be a good starting point for a refactor/rework later.
